test/permutations.jl

```diff
@@ -66,6 +66,16 @@ end
     @test collect(multiset_permutations("", -1)) == Any[]
     @test length(multiset_permutations("aaaaaaaaaaaaaaaaaaaaab", 21)) == 22
 
+    # derangements
+    @test length(collect(derangements(1:4))) == 9
+    @test length(collect(derangements(1:8))) == derangement(8) == 14833
+    @test collect(derangements(Int[])) == [Int[]]
+    @test_broken collect(derangements([1, 1])) == [Int[]] # currently returns Int[]
+    @test_broken collect(derangements([1, 1, 2])) == [Int[]] # currently returns Int[]
```

Comment:
Here for some reason type inference fails and determines incorrectly that the return type should be `Vector{Int}` instead of `Vector{Vector{Int}}`. I'm not sure how to fix it.